lib/dap/filter/geoip2.rb

```diff
@@ -0,0 +1,87 @@
+require 'maxmind/db'
+
+module Dap
+module Filter
+
+require 'dap/utils/misc'
+
+module GeoIP2Library
+  GEOIP_DIRS = [
+    File.expand_path( File.join( File.dirname(__FILE__), "..", "..", "..", "data")),
+    "/var/lib/geoip",
+    "/var/lib/geoip2"
+  ]
+  GEOIP_CITY = %W{ GeoLite2-City.mmdb }
+  GEOIP_ASN = %W{ GeoLite2-ASN.mmdb }
+
+  @@geo_city = nil
+  @@geo_asn = nil
+
+  GEOIP_DIRS.each do |d|
+    GEOIP_CITY.each do |f|
+      path = File.join(d, f)
+      if ::File.exist?(path)
+        @@geo_city = MaxMind::DB.new(path, mode: MaxMind::DB::MODE_MEMORY)
+        break
+      end
+    end
+    GEOIP_ASN.each do |f|
+      path = File.join(d, f)
+      if ::File.exist?(path)
+        @@geo_asn = MaxMind::DB.new(path, mode: MaxMind::DB::MODE_MEMORY)
+        break
+      end
+    end
+  end
+end
+
+
+#
+# Add GeoIP2 tags using the MaxMind GeoIP2::City
+#
+class FilterGeoIP2
```

Comment:
we should name this `geo_ip2_city` to better differentiate between the city, asn and isp databases (follows the PR on godap atm).

Reply:
Fair.  What matters more currently is the legacy ~compatibility which I have yet to add.